src/zombies.test.ts

```diff
@@ -1,30 +1,121 @@
-import { ok } from "node:assert/strict";
+import { ok, equal } from "node:assert/strict";
 import { test } from "node:test";
 
+const numIsValidInteger = (num: unknown): num is number => {
+  return typeof num === 'number' && Number.isInteger(num) && num >= 0;
+};
+
+const isValidString = (name: string): name is string => {
+  return typeof name === 'string' && name.length > 0;
+};
+
+type ZombieReturnObj = {
+  isFull: () => boolean;
+  addZombie: (zombie: string) => void;
+  zombiesInRoom: () => string[];
+  noZombiesInRoom: () => number;
+  spaceLeft: () => number; 
+};
+
 const createRoom = (capacity: number) => {
-  const _capacity = capacity;
+  if (!numIsValidInteger(capacity)) return `Error: Parameter 'capacity' must be a positive integer`;
+  
+  const _capacity = capacity; 
+  let _zombiesInRoom: string[] = [];
 
   return {
-    isFull: () => true,
+    isFull: () => {
+      if (_capacity === 0 || _capacity === _zombiesInRoom.length) return true;      
```

Comment:
When returning true or false in an if-statement, you can usually just return the result of the boolean expressing in the if-statement: `return _capacity === 0 || _capacity === _zombiesInRoom.length`. And the 0 capacity check can be removed since `_capacity === _zombiesInRoom.length` includes that check.

Reply:
Good point!